src/Azure.Functions.Cli/Common/Executable.cs

```diff
@@ -94,48 +97,78 @@ public async Task<int> RunAsync(Action<string> outputCallback = null, Action<str
                 };
                 Process.EnableRaisingEvents = true;
             }
-
             try
             {
                 Process.Start();
+
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    // Ensure child processes are cleaned up
+                    _jobObjectRegistry = new JobObjectRegistry();
+                    _jobObjectRegistry.Register(Process);
+                }
+
                 if (_streamOutput)
                 {
                     Process.BeginOutputReadLine();
                     Process.BeginErrorReadLine();
                 }
-                
+
                 if (!string.IsNullOrEmpty(stdIn))
                 {
                     Process.StandardInput.WriteLine(stdIn);
                     Process.StandardInput.Close();
                 }
-            }
-            catch (Win32Exception ex)
-            {
-                if (ex.Message == "The system cannot find the file specified")
+
+                if (timeout is null)
+                {
+                    return await exitCodeTask.ConfigureAwait(false);
+                }
+                else
                 {
-                    throw new FileNotFoundException(ex.Message, ex);
+                    var completedTask = await Task.WhenAny(exitCodeTask, Task.Delay(timeout.Value)).ConfigureAwait(false);
```

Comment:
what version of .net are we on? Do you have access to [WaitAsync](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.waitasync?view=net-8.0)?

Reply:
I think on .net 8, let me double check and will update if i can!